src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -33,16 +33,52 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
     /// <param name="builder">Distributed application builder</param>
     /// <param name="name">Name of parameter resource</param>
     /// <param name="value">A string value to use for the paramater</param>
+    /// <param name="publishValue">Indicates whether the value should be published to the manifest.</param>
     /// <param name="secret">Optional flag indicating whether the parameter should be regarded as secret.</param>
     /// <returns>Resource builder for the parameter.</returns>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters",
                                                      Justification = "third parameters are mutually exclusive.")]
-    public static IResourceBuilder<ParameterResource> AddParameter(this IDistributedApplicationBuilder builder, string name, string value, bool secret = false)
+    public static IResourceBuilder<ParameterResource> AddParameter(this IDistributedApplicationBuilder builder, string name, string value, bool publishValue = false, bool secret = false)
     {
-        // An alternate implementation is to use some ConstantParameterDefault implementation that always returns the default value.
-        // However, doing this causes a "default": {} to be written to the manifest, which is not valid.
-        // And note that we ignore parameterDefault in the callback, because it can never be non-null, and we want our own value.
-        return builder.AddParameter(name, parameterDefault => value, secret: secret);
+        return builder.AddParameter(name, () => value, publishValue, secret);
+    }
+
+    /// <summary>
+    /// Adds a parameter resource to the application with a value coming from a callback function.
+    /// </summary>
+    /// <param name="builder">Distributed application builder</param>
+    /// <param name="name">Name of parameter resource</param>
+    /// <param name="valueGetter">A callback function that returns the value of the parameter</param>
+    /// <param name="publishValue">Indicates whether the value should be published to the manifest.</param>
+    /// <param name="secret">Optional flag indicating whether the parameter should be regarded as secret.</param>
+    /// <returns>Resource builder for the parameter.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters",
+                                                     Justification = "third parameters are mutually exclusive.")]
+    public static IResourceBuilder<ParameterResource> AddParameter(this IDistributedApplicationBuilder builder, string name, Func<string> valueGetter, bool publishValue = false, bool secret = false)
+    {
+        var val = valueGetter();
```

Comment:
@davidfowl @mitchdenny is this method really meaningful if we are instantly evaluating the callback? How is that any different from users calling the previous overload and evaluating themselves?

And we can't really delay the evaluation, because would happen on the next line anyway.

Reply:
This should be evaluated in side of the lambda, not immediately.

Reply:
Right, but my point is that this doesn't actually delay the evaluation, because it'll happen anyway before this method call returns. So what does that actually buy?

See the stack that comes right back to the lambda:

```
Aspire.Hosting.ParameterResourceBuilderExtensions.AddParameter.AnonymousMethod__0(Aspire.Hosting.ApplicationModel.ParameterDefault parameterDefault) Line 62	C#
Aspire.Hosting.ApplicationModel.ParameterResource.Value.get() Line 39	C#
Aspire.Hosting.ParameterResourceBuilderExtensions.AddParameter(Aspire.Hosting.IDistributedApplicationBuilder builder, string name, System.Func<Aspire.Hosting.ApplicationModel.ParameterDefault, string> callback, bool secret, bool connectionString, Aspire.Hosting.ApplicationModel.ParameterDefault parameterDefault, string configurationKey) Line 148	C#
Aspire.Hosting.ParameterResourceBuilderExtensions.AddParameter(Aspire.Hosting.IDistributedApplicationBuilder builder, string name, System.Func<string> valueGetter, bool publishValue, bool secret) Line 61	C#
```

Reply:
We can change it in the future

Reply:
Ok, I changed it. It added a little bit of complexity to avoid having the lambda evaluated twice.